pyproject.toml

```diff
@@ -99,6 +99,59 @@ skip = [".gitignore", "__init__.py"]
 [tool.flake8]
 ignore = ["F811", "W503", "W504"]
 
+[tool.mypy]
+files = ["torch_frame"]
+install_types = true
+non_interactive = true
+ignore_missing_imports = true
+show_error_codes = true
+warn_redundant_casts = true
+warn_unused_configs = true
+warn_unused_ignores = true
+
+# TODO: the goal is for this ignore list to be empty
+[[tool.mypy.overrides]]
+ignore_errors = true
+# Run this command to generate this list of files
+# mypy --no-error-summary 2>&1 | tr ':' ' ' | awk '{print $1}' | sort | uniq | sed 's/\.py//g; s|src/||g;  s|\/|\.|g' | xargs -I {} echo '"{}",'
+module = [
+    "torch_frame.data.dataset",
+    "torch_frame.data.loader",
```

Comment:
For reviewers, this PR doesn't try to fix any of type annotations, but the goal is to fix annotation errors in all these files in future PRs.